python/az/aro/azext_aro/custom.py

```diff
@@ -229,16 +249,36 @@ def __init__(self, disk_encryption_id, master_subnet_id, worker_subnet_id):
                 new_err = []
                 for txt in err:
                     new_err.append(textwrap.fill(txt, width=160))
-                errors.append(new_err)
+                errors_and_warnings.append(new_err)
+
+    if len(errors_and_warnings) > 0:
+        # Separate errors and warnings into separate arrays
+        warnings=[]
+        errors=[]
+        for issue in errors_and_warnings:
+            if issue[2] == "Warning":
+                warnings.append(issue)
+            else:
+                errors.append(issue)
+    else:
+        logger.info("No validation errors or warnings")
+    
+    if len(warnings) > 0:
+        if len(errors) == 0 and warnings_as_text:
+            full_msg="The following issues will have a minor impact on cluster creation:"
```

Comment:
First, I really like this initiative. Our install process is fairly long and it can be a poor customer experience if the installation fails out due to something avoidable. I do agree with Ghislain in that we may be over-communicating here. What do you think of something more binary? If there are no issues found - proceed with the install as per normal. If blocking issues are present, try to communicate them in a customer friendly way (i.e. reasons they can self-resolve and don't have to necessarily contact support for). We don't want the customer to abandon an install on a warning, I'd rather they get the cluster going asap.

Reply:
It is probably worth making clear that I have not tired to alter the existing errors and warnings they were written and approved by other people and including the PMs.  

And while I have made changes to the functionality of the validation it was only to enable warnings and error types, so that when the `aro_validate` was run from inside the `aro_create` it would not fail on what is clearly a warning.  I would normally call this scope creep and popped this in a separate jira.  For expediency and that it was relatively minor change I chose to include it as part of the current Jira.

I agree (especially with hindsight) the message `The following issues will have a minor impact on cluster creation` **should be improved but at the time I was unable to think of anything better.  Suggestions are welcome.**

Other changes to the functionality of validations may tip the ARO-2887 card too far and it would be better to raise another Jira(s) to remediate and improve the previous work to the existing validations before implementing what should be a simple change of inserting `az_validate` into `az_create`

Cross referencing other [comments made in slack](https://redhat-internal.slack.com/archives/C02ULBRS68M/p1684147023812969)

Reply:
I feel like we need a "styling guide" now for `warn` vs `err` in validation. Does `warn` mean "SRE found an issue, but we can fix it, OK?" and error means "You need to fix something".

As written now, if we agree that's what `warn` means, I'd rather see a message like the following if we only see warnings and no errors:

```
The following issues were found when validating your cluster creation parameters:
{table}
Would you like to proceed? (y/N):
```

As this PR is written, that would currently look something like:

```
The following issues were found when validating your cluster creation parameters:
- resource FOO is missing role BAR for service principal BIZ (These roles will be automatically added during cluster creation)
- resource FOO is missing role BAR for service principal BIZ (These roles will be automatically added during cluster creation)
- resource FOO is missing role BAR for service principal BIZ (These roles will be automatically added during cluster creation)
Would you like to proceed? (y/N):
```

Reply:
- No suggestion or recommendations requested by PMs/BUs in [this slack thread](https://redhat-internal.slack.com/archives/CCV9YF9PD/p1684220507760709).  (This was recommended by Ghislain)


- The confusing addition of the warning to the creation subcommand has been removed.

  > The following issues will have a minor impact on cluster creation:

- Adding the confirmation
  ~~~
  The following issues were found when validating your cluster creation parameters:
  {table}
  Would you like to proceed? (y/N):
  ~~~
  Would change the functionality of the create sub command and potentially break customer scripts as they would then pause/hang indefinitely at a warning.  I dont think that would be an improvement for a customer trying to automate cluster creations.